kotlin/services/ec2/src/main/kotlin/com/kotlin/ec2/EC2Scenario.kt

```diff
@@ -0,0 +1,528 @@
+// Get a list of instance types.
+suspend fun getInstanceTypesSc(): String {
+    var instanceType = ""
+    val filterObs = ArrayList<Filter>()
+    val filter = Filter {
+        name = "processor-info.supported-architecture"
+        values = listOf("arm64")
+    }
+
+    filterObs.add(filter)
+    val typesRequest = DescribeInstanceTypesRequest {
+        filters = filterObs
+        maxResults = 10
+    }
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeInstanceTypes(typesRequest)
+        response.instanceTypes?.forEach { type ->
+            println("The memory information of this type is ${type.memoryInfo?.sizeInMib}")
+            println("Network information is ${type.networkInfo}")
+            instanceType = type.instanceType.toString()
+        }
+        return instanceType
+    }
+}
+// snippet-end:[ec2.kotlin.describe_instance_type.main]
+
+// Display the Description field that corresponds to the instance Id value.
+suspend fun describeImageSc(instanceId: String): String? {
+    val imagesRequest = DescribeImagesRequest {
+        imageIds = listOf(instanceId)
+    }
+
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeImages(imagesRequest)
+        println("The description of the first image is ${response.images?.get(0)?.description}")
+        println("The name of the first image is  ${response.images?.get(0)?.name}")
+
+        // Return the image id value.
+        return response.images?.get(0)?.imageId
+    }
+}
+
+// Get the Id value of an instance with amzn2 in the name.
+suspend fun getParaValuesSc(): String? {
+    val parameterRequest = GetParametersByPathRequest {
+        path = "/aws/service/ami-amazon-linux-latest"
+    }
+
+    SsmClient { region = "us-west-2" }.use { ssmClient ->
+        val response = ssmClient.getParametersByPath(parameterRequest)
+        response.parameters?.forEach { para ->
+            println("The name of the para is: ${para.name}")
+            println("The type of the para is: ${para.type}")
+            if (para.name?.let { filterName(it) } == true) {
+                return para.value
+            }
+        }
+    }
+    return ""
+}
+
+fun filterName(name: String): Boolean {
+    val parts = name.split("/").toTypedArray()
+    val myValue = parts[4]
+    return myValue.contains("amzn2")
+}
+
+suspend fun describeSecurityGroupsSc(groupId: String) {
+    val request = DescribeSecurityGroupsRequest {
+        groupIds = listOf(groupId)
+    }
+
+    Ec2Client { region = "us-west-2" }.use { ec2 ->
+        val response = ec2.describeSecurityGroups(request)
+        for (group in response.securityGroups!!) {
+            println("Found Security Group with id " + group.groupId.toString() + " and group VPC " + group.vpcId)
+        }
+    }
+}
+
+suspend fun createEC2SecurityGroupSc(groupNameVal: String?, groupDescVal: String?, vpcIdVal: String?): String? {
+    val request = CreateSecurityGroupRequest {

```

Comment:
In the spec it says it should add the user's IP to the security group, but I don't see it getting or requesting the user's IP to add.

Reply:
WIll look into this